tools/cmake/app.cmake

```diff
@@ -9,11 +9,9 @@ endif()
 
 include(${MBED_CONFIG_PATH}/mbed_config.cmake)
 
-set(MBED_TOOLCHAIN_FILE_USED FALSE)
-
 # Set default toolchain file
-if(NOT CMAKE_TOOLCHAIN_FILE)
-    set(MBED_TOOLCHAIN_FILE_USED TRUE)
+if(NOT CMAKE_TOOLCHAIN_FILE OR MBED_TOOLCHAIN_FILE_USED)
+    set(MBED_TOOLCHAIN_FILE_USED TRUE CACHE BOOL "")
```

Comment:
should it be internal ?

Reply:
Should it? I do not know.

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/3525f74c23a1148f46aca37d8c2ca20969d9b94b..a70edcda0ab4c1c7a2b046a48540d87271eb8edc).